pkg/tessera/note.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2025 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Heavily borrowed from https://gist.githubusercontent.com/AlCutter/c6c69076dc55652e2d278900ccc1a5e7/raw/aac2bafc17a8efa162bd99b4453070b724779307/ecdsa_note.go - thanks, Al
+
+package tessera
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/binary"
+	"fmt"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+	"golang.org/x/mod/sumdb/note"
+)
+
+// noteSigner uses an arbitrary sigstore signer to implement golang.org/x/mod/sumdb/note.Signer,
+// which is used in Tessera to sign checkpoints in the signed notes format
+// (https://github.com/C2SP/C2SP/blob/main/signed-note.md).
+type noteSigner struct {
```

Comment:
Should we call this an ECDSA note signer? From [C2SP](https://github.com/C2SP/C2SP/blob/main/signed-note.md), the key ID calculation is for the `0x02` signature type. 

This is https://github.com/sigstore/rekor/issues/2062. 

We can use Tessera's note signer if the key is ed25519, and this signer if the key is ecdsa.

If it's RSA, we need to define our own key ID since that is not defined by C2SP. I'd suggest `SHA256(key name || 0xA || 0xFF || "RSA-PKCS#1v1.5" || public key)[:4]`. 0xFF comes from the spec, and we include an additional identifier recommended by the spec.